experimental/Origami/src/Origami.jl

```diff
@@ -0,0 +1,299 @@
+    # TODO check for transitivity? GAP already does this and this was so slow
+    # for huge origamis
+    #perm_group = permutation_group(deg, [h, v])
+    #if transitivity(perm_group) > 0
+        
+    #end
+end
+
+function origami_disconnected(h::PermGroupElem, v::PermGroupElem, d::Integer)
+    return Origami(GAP.Globals.OrigamiNC(GapObj(h), GapObj(v), d), h, v, d)
+end
+
+function Base.:(==)(a::Origami, b::Origami)
+    # TODO rewrite this? for now use Gap equality
+    return GAP.evalstr("$(string(a)) = $(string(b))")
+end
+
+function horizontal_perm(o::Origami)
+    return o.h
+end
+
+function vertical_perm(o::Origami)
+    return o.v
+end
+
+function degree(o::Origami)
+    return o.d
+end
+
+function Base.show(io::IO, o::Origami)
+    h = horizontal_perm(o)
+    v = vertical_perm(o)
+    d = degree(o)
+    print(io, "Origami ($(h),$(v), $(d))")
+end
+
+GapObj(O::Origami) = O.o
+
+function stratum(o::Origami)
+    # cannot use comm from Oscar because of different definitions
+    h = horizontal_perm(o)
+    v = vertical_perm(o)
+    commutator = h * v * h^-1 * v^-1
+    cycle_struc = cycle_structure(commutator)
+    different_cycle_count = length(cycle_struc)
+    stratum::Vector{Integer} = []
+    for i in 1:different_cycle_count
+        cycle_type_entry = cycle_struc[i]
+        cycle_type_length = cycle_struc[i][1]
+        if(cycle_type_entry[1] == 1)
+            # ignore cycles of length 1
+            continue
+        end
+        for j in 1:cycle_type_entry[2]
+            push!(stratum, ZZ(cycle_type_length - 1))
+        end
+    end
+    return stratum
+end
+
+function genus(o::Origami)
+    return ZZ((sum(stratum(o)) + 2) * 0.5)
+end
+
+function veech_group(O::Origami)
+    # TODO use hashtables or not? Implement modular subgroup?
+    GAP.Globals.ComputeVeechGroupWithHashTables(GapObj(O))
+end
+
+function index_monodromy_group(o::Origami)
+    GAP.Globals.IndexOfMonodromyGroup(GapObj(o))
+end
+
+function sum_of_lyapunov_exponents(o::Origami)
+    gap_obj = GAP.Globals.SumOfLyapunovExponents(GapObj(o))
+    return GAP.gap_to_julia(gap_obj)
+end
+
+function translations(o::Origami)
+    h = horizontal_perm(o)
+    v = vertical_perm(o)
+    G = normalform_conjugators(o)
+
+    O = (i -> origami(i^-1 * h * i, i^-1 * v * i)).(G)
+    list = [cperm([degree(o)])]
+
+    l = length(O)
+    for i in 1:l
+        positions = findall(item -> item == O[i], O)
+        if length(positions) != 1
+            for j in positions
+                push!(list, G[i] * G[j]^-1)
+            end
+        end
+    end
+
+    return collect(Set(list))
+end
+
+function is_hyperelliptic(o::Origami)
+    # check whether -1 is in the veech group
+	if !veech_group_is_even(o) 
+        return false
+    end
+
+    n = 2
+    n_inv = 1 / 2
+	x = horizontal_perm(o)
+	y = vertical_perm(o)
+	g = genus(o)
+
+    L = point_reflections(o)
+	L = filter(i -> order(i) == 2, L)
+
+	if L == []
```

Comment:
The use of `[]` creates an object, just to check if `L` is empty. Better:
```suggestion
	if isempty(L)
```